### PR TITLE
MGDOBR-476 Implement test for periodically checking of long running…

### PR DESCRIPTION
**Pull request by sutaakar**
… bridge instance

Signed-off-by: Karel Suta <ksuta@redhat.com>

[MGDOBR-476](https://issues.redhat.com/browse/MGDOBR-476)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: .github/workflows/dev-cluster-long-running-bridge.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          echo "::add-mask::${{ secrets.CI_MAIL_PASSWORD }}"
+      - name: Java and Maven Setup
+        uses: ./.github/actions/java-maven-setup
+        with:
+          cache-key-prefix: ${{ runner.os }}
+      - name: Compile sandbox repo
+        uses: ./.github/actions/maven
+        with:
+          maven-command: clean install -Dquickly
+      - name: Run integration tests
```
Comment (jiripetrlik) [inline .github/workflows/dev-cluster-long-running-bridge.yml:34]: Same as with job name. Run periodical slack check would be probably better.

**Review by jiripetrlik — COMMENTED** [review]
I have one minor comment.

**Comment by sutaakar** [comment]
@jiripetrlik Good catch, should be addressed now.

**Review by jiripetrlik — APPROVED** [review]

**Comment by sutaakar** [comment]
@Srihari1192 @r00ta Can you please take a look?

File: .github/workflows/dev-cluster-long-running-bridge.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        run: |
+          echo "::add-mask::${{ secrets.DEV_CLUSTER_MANAGER_URL }}"
+          echo "::add-mask::${{ secrets.DEV_SSO_CLIENT_ID }}"
+          echo "::add-mask::${{ secrets.DEV_SSO_CLIENT_SECRET }}"
+          echo "::add-mask::${{ secrets.SLACK_WEBHOOK_URL }}"
+          echo "::add-mask::${{ secrets.SLACK_WEBHOOK_TOKEN }}"
+          echo "::add-mask::${{ secrets.SLACK_CHANNEL }}"
+          echo "::add-mask::${{ secrets.CI_MAIL_USERNAME }}"
+          echo "::add-mask::${{ secrets.CI_MAIL_PASSWORD }}"
+          echo "::add-mask::${{ secrets.REMOTE_CLUSTER_LONG_RUNNING_BRIDGE_ENDPOINT }}"
```
Comment (r00ta) [inline .github/workflows/dev-cluster-long-running-bridge.yml:26]: You still have to create it right?

Comment (sutaakar) [reply .github/workflows/dev-cluster-long-running-bridge.yml:26]: Yes, I plan to create it once reviewing process is done.

File: .github/workflows/dev-cluster-long-running-bridge.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          DEV_CLUSTER_MANAGER_URL: ${{ secrets.DEV_CLUSTER_MANAGER_URL }}
+          DEV_SSO_CLIENT_ID: ${{ secrets.DEV_SSO_CLIENT_ID }}
+          DEV_SSO_CLIENT_SECRET: ${{ secrets.DEV_SSO_CLIENT_SECRET }}
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+          SLACK_WEBHOOK_TOKEN: ${{ secrets.SLACK_WEBHOOK_TOKEN }}
+          SLACK_CHANNEL: ${{ secrets.SLACK_CHANNEL }}
+          REMOTE_CLUSTER_LONG_RUNNING_BRIDGE_ENDPOINT: ${{ secrets.REMOTE_CLUSTER_LONG_RUNNING_BRIDGE_ENDPOINT }}
+        uses: ./.github/actions/maven
+        with:
+          maven-command: clean verify -Pperiodical-slack-check -Dparallel -Devent-bridge.manager.url=$DEV_CLUSTER_MANAGER_URL -Dkeycloak.realm.url=https://sso.redhat.com/auth/realms/redhat-external -Dbridge.client.id=$DEV_SSO_CLIENT_ID -Dbridge.client.secret=$DEV_SSO_CLIENT_SECRET -Dremote.cluster.bridge.endpoint=$REMOTE_CLUSTER_LONG_RUNNING_BRIDGE_ENDPOINT
```
Comment (jomarko) [inline .github/workflows/dev-cluster-long-running-bridge.yml:46]: Do I understand correctly that this profile run single test? Is there any plan to run all tests? Or other way, could you please clarify why we picked up this test for running against long running bridge instance? 
```
    <profile>
      <id>periodical-slack-check</id>
      <build>
        <plugins>
          <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-failsafe-plugin</artifactId>
            <executions>
              <execution>
                <goals>
                  <goal>integration-test</goal>
                  <goal>verify</goal>
                </goals>
              </execution>
            </executions>
            <configuration>
              <includes>
                <include>**/RunPeriodicalSlackActionSuite.java</include>
              </includes>
            </configuration>
          </plugin>
        </plugins>
      </build>
    </profile>
```

If that is a case, it seems to me we set non used environment variables like webhook

**Review by jomarko — COMMENTED** [review]
One question @sutaakar 

Do I understand correctly that this profile run single test? Is there any plan to run all tests? Or other way, could you please clarify why we picked up this test for running against long running bridge instance? 
```
    <profile>
      <id>periodical-slack-check</id>
      <build>
        <plugins>
          <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-failsafe-plugin</artifactId>
            <executions>
              <execution>
                <goals>
                  <goal>integration-test</goal>
                  <goal>verify</goal>
                </goals>
              </execution>
            </executions>
            <configuration>
              <includes>
                <include>**/RunPeriodicalSlackActionSuite.java</include>
              </includes>
            </configuration>
          </plugin>
        </plugins>
      </build>
    </profile>
```

**Comment by sutaakar** [comment]
@jomarko Purpose of this testing is to make sure that the configured Bridge/Processor and their connection is available and stable over long period of time.
Also the test verifies that changes done in dev cluster don't break existing deployments.

I put this test separately as its scope is different from other test (which are used to verify the API functionality).

**Comment by jomarko** [comment]
@sutaakar I understand more now. Thank you. I understand there is some connection configured for slack action. 

I understand that probably webhook action is not under a risk, however maybe send to bridge should be then also a subject of similar test?

**Comment by sutaakar** [comment]
@jomarko May be, feel free to create a JIRA and/or implement.

**Review by jomarko — APPROVED** [review]

**Pull request merged**